nncf/pruning/filter_pruning/algo.py

```diff
@@ -83,19 +98,170 @@ def statistics(self, quickly_collected_only=False):
+
+        for h in hook_list:
+            h.remove()
+
+    def _calculate_flops_in_pruned_model(self, modules_in_channels, modules_out_channels):
+        """
+        Calculates number of flops in model with number of input/output channels for nodes from modules_in_channels,
+        modules_out_channels. It allows to count the number of flops in pruned model (with changed number of
+        input/output channels for some nodes).
+        Number of flops calculates as follows: for nodes that isn't in in/out channels used full number of flops from
+        self.nodes_flops.
+        For nodes with keys from in/out channels dicts flops
+        = modules_in_channels[node_id] * modules_out_channels[node_id] * self.nodes_flops_cost[node_id]
+        :param modules_in_channels: numbers of input channels in nodes
+        :param modules_out_channels: numbers of output channels in model
+        :return: number of flops in model
+        """
+        flops = 0
+        graph = self._model.get_original_graph()
+        for nncf_node in graph.get_all_nodes():
+            if nncf_node.node_id in modules_in_channels:
+                flops += int(modules_in_channels[nncf_node.node_id] * modules_out_channels[nncf_node.node_id] * \
+                         self.nodes_flops_cost[nncf_node.node_id])
+            elif nncf_node.node_id in self.nodes_flops:
+                flops += self.nodes_flops[nncf_node.node_id]
+        return flops
+
+    def _calculate_flops_in_uniformly_pruned_model(self, pruning_rate):
+        """
+        Prune all prunable modules in model with pruning_rate rate and returns flops of pruned model.
+        :param pruning_rate: proportion of zero filters in all modules
+        :return: flops number in pruned model
+        """
+        tmp_in_channels = self.modules_in_channels.copy()
+        tmp_out_channels = self.modules_out_channels.copy()
+
+        for group in self.pruned_module_groups_info.get_all_clusters():
+            assert all([tmp_out_channels[group.nodes[0].nncf_node_id] == tmp_out_channels[node.nncf_node_id] for node in
+                        group.nodes])
+            # prune all nodes in cluster (by output channels)
+            old_out_channels = self.modules_out_channels[group.nodes[0].nncf_node_id]
+            num_of_sparse_elems = get_rounded_pruned_element_number(old_out_channels, pruning_rate)
+            new_out_channels_num = old_out_channels - num_of_sparse_elems
+
+            for node in group.nodes:
+                tmp_out_channels[node.nncf_node_id] = new_out_channels_num
+
+            # Prune in_channels in all next nodes of cluster
+            next_nodes = self.next_nodes[group.id]
+            for node_id in next_nodes:
+                tmp_in_channels[node_id] = new_out_channels_num
+        flops = self._calculate_flops_in_pruned_model(tmp_in_channels, tmp_out_channels)
+        return flops
+
+    def _find_layerwise_pruning_rate(self, target_flops_pruning_rate):
```

Comment:
@AlexKoff88 Please, take a closer look at this algo of layer-wise pruning rate search for FLOPs pruning target (quickly discussed earlier).

Reply:
LGTM